test/unit/attributes.js

```diff
@@ -4,127 +4,306 @@ import helpers from '../lib/helpers';
+              created = true;
+              data.newValue.should.equal('created');
+              element.setAttribute('open', 'updated');
+            },
+            updated: function (element, data) {
+              updated = true;
+              data.oldValue.should.equal('created');
+              data.newValue.should.equal('updated');
+              element.removeAttribute('open');
+            },
+            removed: function (element, data) {
+              created.should.equal(true);
+              updated.should.equal(true);
+              data.oldValue.should.equal('updated');
+              done();
+            }
+          }
+        }
+      });
+
+      helpers.fixture('<my-el open="created"></my-el>');
+    });
+
+    it('should accept a function insead of an object for a particular attribute definition.', function (done) {
+      skate('div', {
+        attributes: {
+          open: function (element, data) {
+            if (data.type === 'created') {
+              element.setAttribute('open', 'updated');
+            } else if (data.type === 'updated') {
+              element.removeAttribute('open');
+            } else if (data.type === 'removed') {
+              assert(true);
+              done();
+            }
+          }
+        }
+      });
+
+      helpers.fixture('<div id="attrtest" open="created"></div>');
+    });
+
+    it('should accept a function insead of an object for the entire attribute definition.', function (done) {
+      skate('div', {
+        attributes: function (element, data) {
           if (data.type === 'created') {
-            element.setAttribute('open', 'updated');
+            setTimeout(function () {
+              element.setAttribute('open', 'updated');
+            });
           } else if (data.type === 'updated') {
-            element.removeAttribute('open');
+            setTimeout(function () {
+              element.removeAttribute('open');
+            });
           } else if (data.type === 'removed') {
             assert(true);
             done();
           }
         }
-      }
-    });
+      });
 
-    helpers.fixture('<div id="attrtest" open="created"></div>');
-  });
+      helpers.fixture('<div id="attrtest" open="created"></div>');
+    });
 
-  it('should accept a function insead of an object for the entire attribute definition.', function (done) {
-    skate('div', {
-      attributes: function (element, data) {
-        if (data.type === 'created') {
-          setTimeout(function () {
-            element.setAttribute('open', 'updated');
-          });
-        } else if (data.type === 'updated') {
-          setTimeout(function () {
-            element.removeAttribute('open');
-          });
-        } else if (data.type === 'removed') {
-          assert(true);
-          done();
+    it('should allow a catchall callback to be specified that catches all changes (same as passing a function instead of an object)', function (done) {
```

Comment:
There should probably be a test that makes sure `catchall` is called only for the undefined lifecycle events.